Libraries/LibJS/Runtime/StringPrototype.cpp

```diff
@@ -507,6 +520,71 @@ JS_DEFINE_NATIVE_FUNCTION(StringPrototype::slice)
     return js_string(vm, string_part);
 }
 
+JS_DEFINE_NATIVE_FUNCTION(StringPrototype::split)
+{
+    // FIXME Implement the @@split part
+
+    auto string = ak_string_from(vm, global_object);
+    if (string.is_null())
+        return {};
+
+    auto* result = Array::create(global_object);
+    size_t result_len = 0;
+
+    auto limit = static_cast<u32>(MAX_U32);
+    if (!vm.argument(1).is_undefined())
+        limit = vm.argument(1).to_u32(global_object);
```

Comment:
The `to_number()` part in `to_u32()` can fail so this needs an exception check:

```suggestion
    if (!vm.argument(1).is_undefined()) {
        limit = vm.argument(1).to_u32(global_object);
        if (vm.exception())
            return {};
    }
```

Reply:
As a general rule of thumb I figured out while messing with these in Spreadsheet, _anything_ that takes a global object can (and likely will) throw.

Reply:
That, as well as the `to_*` part (as opposed to `as_*`) :)

Reply:
I was thinking, maybe we should somehow mark the methods that may throw? Coming back to LibJS after some time I realized that it has grown beyond the capabilities of my tiny brain.

Reply:
The closest I can think of is `[[nodiscard]]`, but that won't help as exceptions are side effects...

Reply:
Forgot that you can't define custom attributes in C++. :(